src/transforms/projectionpursuit.jl

```diff
@@ -0,0 +1,201 @@
+# ------------------------------------------------------------------
+# Licensed under the MIT License. See LICENSE in the project root.
+# ------------------------------------------------------------------
+
+"""
+    ProjectionPursuit(;tol=1e-6, maxiter=100, deg=5, perc=.9, n=100)
+
+The projection pursuit multivariate transform converts any multivariate distribution into
+the standard multivariate Gaussian distribution.
+
+This iterative algorithm repeatedly finds a direction of projection `α` that maximizes a score of
+non-Gaussianity known as the projection index `I(α)`. The samples projected along `α` are then
+transformed with the [`Quantile`](@ref) transform to remove the non-Gaussian structure. The
+other coordinates in the rotated orthonormal basis `Q = [α ...]` are left untouched.
+
+The non-singularity of Q is controlled by assuring that norm(det(Q)) ≥ `tol`. The iterative 
+process terminates whenever the transformed samples are "more Gaussian" than `perc`% of `n`
+randomly generated samples from the standard multivariate Gaussian distribution, or when the 
+number of iterations reaches a maximum `maxiter`.
+
+# Examples
+
+```julia
+ProjectionPursuit()
+ProjectionPursuit(deg=10)
+ProjectionPursuit(perc=.85, n=50)
+ProjectionPursuit(tol=1e-4, maxiter=250, deg=5, perc=.95, n=100)
+```
+
+"""
+
+struct ProjectionPursuit{T} <: StatelessFeatureTransform
+  tol::T
+  maxiter::Int
+  deg::Int
+  perc::T
+  n::Int
+end
+
+ProjectionPursuit(;tol=1e-6, maxiter=100, deg=5, perc=.95, n=100) =
+  ProjectionPursuit{typeof(tol)}(tol, maxiter, deg, perc, n)
+
+isrevertible(::Type{<:ProjectionPursuit}) = true
+
+# transforms a row of random variables into a convex combination 
+# of random variables with values in [-1,1] and standard normal distribution
+function rscore(Z, α)
+  ᾱ = (1/norm(α)) .* α
+  X = Z * ᾱ
+  2 .* cdf.(Normal(), X) .- 1
+end
+
+# projection index of sample along a given direction
+function pindex(transform, Z, α)
+  d = transform.deg
+  X = rscore(Z, α)
```

Comment:
Should this variable be called `r` instead? It is the output of the rscore function, correct? Also, it seems like it is a vector, we try to use uppercase letters for matrices and lowercase for vectors.

Reply:
It is a vector, but in the paper they call it X, so I used the same notation.

Reply:
![image](https://user-images.githubusercontent.com/3345261/193113917-c9ff9762-6ae8-4cb5-a3a5-7cc7af49aa61.png)

Reply:
@DianaPat please follow the notation in the paper. As shown in the screenshot above the variable is called R.